UI/src/main/java/AdsServlet.java

```diff
@@ -0,0 +1,57 @@
+import adList.Ad;
+import adList.AdList;
+import com.google.gson.Gson;
+
+import javax.servlet.*;
+import javax.servlet.http.*;
+import javax.servlet.annotation.*;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+@WebServlet(name = "AdsServlet", value = "/ads")
+public class AdsServlet extends HttpServlet {
+    AdList adList;
+    @Override
+    public void init() throws ServletException {
+        adList = new AdList();
+        List<String> tags1 = new ArrayList<>();
+        tags1.add("tags");
+        Ad ad = new Ad("1","description", new Date(),"link", "vendor",tags1,"discount", new Date());
+        adList.add(ad);
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        String id = request.getParameter("id");
+        if(id != null){
+            response.getWriter().println("<html>" + (adList != null ? adList.get(id).toString() : "null") + "</html>");
```

Comment:
1) Consider using `String.format()` instead of the concatenation
2) The HTML markup won't be valid unless `<DOCTYPE HTML>`, `<head>`, `<title>` and `<body>` tags are present. You can always check with https://validator.w3.org/
2) Is it worth reporting word "null" to the user? What way it would help him?) Consider outputting some specific status code instead.